File: app/src/main/java/com/silvertak/relationshipsmanager/customInterface/ThreadCallBack.java
package com.silvertak.relationshipsmanager.customInterface;

import com.silvertak.relationshipsmanager.data.CallLogInfo;
import com.silvertak.relationshipsmanager.data.ContactInfo;
import com.silvertak.relationshipsmanager.data.PersonRelationshipInfo;

import java.util.ArrayList;

public interface ThreadCallBack {
    void onContactMappingComplete(ArrayList<ContactInfo> contactInfos);
    void onCallLogMappingComplete(ArrayList<CallLogInfo> callLogInfos);
    void onPersonRelationshipMappingComplete(ArrayList<PersonRelationshipInfo> personRelationshipInfos);
}

File: app/src/main/java/com/silvertak/relationshipsmanager/customInterface/ThreadCallBack.kt
package com.silvertak.relationshipsmanager.customInterface

import com.silvertak.relationshipsmanager.vo.CallLogVO
import com.silvertak.relationshipsmanager.vo.ContactInfoVO
import com.silvertak.relationshipsmanager.vo.PersonRelationshipInfo
import java.util.*

interface ThreadCallBack {
    fun onContactMappingComplete(contactInfos: ArrayList<ContactInfoVO?>?)
    fun onCallLogMappingComplete(callLogInfos: ArrayList<CallLogVO?>?)
    fun onPersonRelationshipMappingComplete(personRelationshipInfos: ArrayList<PersonRelationshipInfo?>?)
}